test/FunctionalTests/Microsoft.OData.Core.Tests/JsonLight/ODataJsonLightBatchWriterTests.cs

```diff
@@ -405,6 +409,112 @@ public async Task WriteBatchResponseWithChangesetAsync()
                 result);
         }
 
+        [Fact]
+        public async Task WriteBatchResponseAsync_WithStreamCopy()
+        {
+            var result = await SetupJsonLightBatchWriterAndRunTestAsync(
+                async (jsonLightBatchWriter) =>
+                {
+                    await jsonLightBatchWriter.WriteStartBatchAsync();
+
+                    var customerOperationMessage = await jsonLightBatchWriter.CreateOperationResponseMessageAsync("1");
+                    customerOperationMessage.SetHeader("odata-version", "4.0");

```

Comment:
Not sure these `SetHeader` lines are necessary but if you choose to retain them, maybe change `odata-version` to `OData-Version` since that's what's defined in https://github.com/OData/odata.net/blob/056e8fdb4cdc1ae21a872fdbd5847c527a079b3e/src/Microsoft.OData.Core/ODataConstants.cs#L55